v_3_4_0/master_template.go

```diff
@@ -2090,6 +2090,10 @@ coreos:
       --node-labels="node-role.kubernetes.io/master,role=master,ip=${DEFAULT_IPV4},{{.Cluster.Kubernetes.Kubelet.Labels}}" \
       --kube-reserved="cpu=200m,memory=250Mi" \
       --system-reserved="cpu=150m,memory=250Mi" \
+      --eviction-soft='memory.available<500Mi' \
+      --eviction-hard='memory.available<350Mi' \
+      --eviction-soft-grace-period='memory.available=15s' \
```

Comment:
Is `500Mi` with `15s` a good combination for soft thresholds when some workloads can hog available resources very quickly? I can imagine the time taken from `500Mi` -> `350Mi` is very short and `eviction-soft-grace-period` states following:
> eviction-soft-grace-period describes a set of eviction grace periods (e.g. memory.available=1m30s) that correspond to how long a soft eviction threshold must hold before triggering a Pod eviction.

This makes me think if `soft-grace-period` of `5s` or even `3s` would be better? I don't have hard evidence, but I'd claim that most of the cases that will hit this are from use of various runtimes with GC and in those cases it is totally exceptional to release allocated memory at least in any meaningful period of time.

To summarize: I'm slightly afraid that time from `500Mi` -> `350Mi` is very short and therefore in most cases kubelet starts hard eviction practically immediately and given conditions, it depends on workload if kubelet gets to a point of selecting the right POD to terminate. If there are processes that use a lot of memory and I/O but which have even remotely reasonable requests values (ElasticSearch or Nexus indexing), it could be that those processes slow down the whole machine so that we would still get occasional flapping NodeStatus / KubeletDown alerts.

Nevertheless, disclaimer: These are just thoughts that came from the values & k8s docs and I don't have any hard evidence backing what I'm saying and therefore I can be totally wrong here. Just 2 cents :)

Reply:
sure, this period is something to consider, I started with the example value of 90s but that felt way too long so I decreased it, but I wasn't sure where I should stop.

5 sec might not be such bad idea.

Reply:
Please don't go this low. Normal application need a bunch of time to properly shut down after receiving SIGTERM. By default afaik kubernetes has graceful termination timeout 30 seconds. May be let's just use 30sec too.

Reply:
We are talking about the grace period for soft threshold, that means that the memory has to be under the limit for this period.

shutdown pod grace period is still same and set to 60s for now

Reply:
https://github.com/giantswarm/k8scloudconfig/pull/390/files#diff-c968cf84270a51fc1f48f59c2f86f053R2096